flytepropeller/pkg/controller/nodes/array/handler.go

```diff
@@ -460,6 +547,18 @@
 
 // Setup handles any initialization requirements for this handler
 func (a *arrayNodeHandler) Setup(_ context.Context, _ interfaces.SetupContext) error {
+	// start workers
+	for i := 0; i < config.GetConfig().NodeExecutionWorkerCount; i++ {
+		worker := worker{
+			gatherOutputsRequestChannel: a.gatherOutputsRequestChannel,
+			nodeExecutionRequestChannel: a.nodeExecutionRequestChannel,
+		}
+
+		go func() {
```

Comment:
Great point. I added this in the worker execution so that we can return the error, similar to how this is [done in workflow execution](https://github.com/flyteorg/flyte/blob/performance%2Farraynode-parallel-workers/flytepropeller/pkg/controller/handler.go#L131-L149).